### DISCUSSION TITLE
Readiness of "on cluster" build functionality

### DISCUSSION THREAD

**Discussion by grafvonb** [knative/func#1313] [General]
I tried today to setup (after successful setup of an AWS cluster with Knative and Tekton) "on cluster" build according to this document: https://github.com/knative/func/blob/main/docs/reference/on_cluster_build.md but it seems to be outdated. Trying to run `func build --builder=s2i` gives strange errors like `error: the 'go' runtime does not define a default 's2i' builder image`. Is "on cluster" option runnable at all? I have used `func` 1.7.

**Comment by zroubalik** [comment]
Hi, s2i build currently doesn't support Go runtime,  Buildpacks approach works. We are still working the on cluster build, should be more mature more.

**Comment by lkingland** [comment]
Our forthcoming release will support Go functions in on-cluster builds, as does the S2I builder.

Also included are support for instance-based method signatures and lifecycle hooks for on-start, on-stop, readiness check, and liveness check.
